src/stream/consumption.ts

```diff
@@ -133,4 +134,92 @@ export class StreamConsumption<T, E> extends StreamBase {
+     *
+     * @see {@link Stream#serialize} if the stream values should be serialized to json
+     * @group Consumption
+     */
+    toReadable(kind: "raw" | "object", options?: { single?: boolean; atoms?: boolean }): Readable;
+
+    /**
+     * Produce a readable node stream with the raw values from the stream.
+     *
+     * @param options.single - Whether to emit only the first atom
+     *
+     * @see {@link Stream#serialize} if the stream values should be serialized to json
+     * @group Consumption
+     */
+    toReadable(kind: "raw", options?: { single?: boolean }): Readable;
+
+    /**
+     * Produce a readable node stream with the values from the stream
+     *
+     * @param options.single - Whether to emit only the first atom
+     * @param options.atoms - By default, only `ok` values are emitted, however enabling this
+     * will emit all values.
+     *
+     * @see {@link Stream#serialize} if the stream values should be serialized to json
+     * @group Consumption
+     */
+    toReadable(kind: "object", options?: { single?: boolean; atoms?: boolean }): Readable;
+
+    toReadable(
+        kind: "raw" | "object",
+        options: { single?: boolean; atoms?: boolean } = {},
+    ): Readable {
+        // Set up a new readable stream that does nothing
+        const s = new Readable({
+            read() {},
+            objectMode: kind === "object",
+        });
+
+        // Spin off asynchronously so that the stream can be immediately returned
+        (async () => {
+            let sentItems = 0;
+
+            for await (const atom of this) {
+                // Determine whether non-ok values should be filtered out
+                if (options?.atoms !== true && !isOk(atom)) {
+                    continue;
+                }
+
+                // Monitor for multiple values being sent when only one is desired
+                if (sentItems > 0 && options?.single) {
+                    console.warn(
+                        "indicated that stream would emit a single value, however multiple were emitted (ignoring)",
+                    );
+                    break;
+                }
+
+                // monitor for non raw values when not using object mode
+                if (
+                    kind === "raw" &&
+                    !(typeof atom.value === "string" || atom.value instanceof Buffer)
+                ) {
+                    s.emit(
+                        "error",
+                        new Error(
+                            `Stream indicated it would emit raw values but emitted a '${typeof atom.value}' object`,
+                        ),
+                    );
+                    break;
+                }
+
+                // Emit atom or atom value
+                s.push(options?.atoms ? atom : atom.value);
+                sentItems += 1;
+            }
+
+            // End the stream
+            s.push(null);
```

Comment:
Include a note about `null` being in the windpipe stream in the doc comment. Pretty much, if someone has a stream of `[ok('hi'), ok(null), ok('asdf')]`, only 'hi' will actually get emitted on to the stream. Will be an edge-case, maybe even worthwhile including a console.log warning if it's detected...